Library/Homebrew/cmd/upgrade.rb

```diff
@@ -92,18 +93,46 @@ def upgrade
       end
     end
 
+    broken_formulae = []
+
     formulae_to_install.each do |f|
       Migrator.migrate_if_needed(f)
       begin
         upgrade_formula(f)
         next if !ARGV.include?("--cleanup") && !ENV["HOMEBREW_UPGRADE_CLEANUP"]
```

Comment:
this cleanup logic is preventing the new linkage-checking code from being run in my test (since I haven't been passing --cleanup)

we aren't going to require `--cleanup` for the linkage checking are we?

Reply:
> we aren't going to require --cleanup for the linkage checking are we?

Yeh, we shouldn't 👍

Reply:
Refactored this part into a new function.